src/graph/executor/query/TraverseExecutor.cpp

```diff
@@ -164,27 +160,18 @@ folly::Future<Status> TraverseExecutor::handleResponse(RpcResponse&& resps) {
   auto listVal = std::make_shared<Value>(std::move(list));
   auto iter = std::make_unique<GetNeighborsIter>(listVal);
   if (currentStep_ == 1) {
-    if (range_ && range_->min() == 0) {
-      result_.rows = buildZeroStepPath(iter.get());
-    }
-    // match (v)-[e:Rel]-(v1:Label1)-[e1*2]->() where id(v0) in [6, 23] return v1
-    // the attributes of v1 will be obtained in the second traverse operator
-    // If the conditions are not met, the path in the previous step needs to be filtered out
-    std::unordered_set<Value, VertexHash, VertexEqual> existVids;
-    existVids.reserve(iter->numRows());
+    initVertices_.reserve(iter->numRows());
     auto vertices = iter->getVertices();
+    // match (v)-[e:Rel]-(v1:Label1)-[e1*2]->() where id(v0) in [6, 23] return v1
+    // save the vertex that meets the filter conditions as the starting vertex of the current
+    // traverse
     for (auto& vertex : vertices.values) {
       if (vertex.isVertex()) {
-        existVids.emplace(vertex);
+        initVertices_.emplace_back(vertex);
```

Comment:
Isn't it necessary to deduplicate the vids for next traverse ?

Reply:
the first column vid in the result returned by getNeighbor is not repeated

Reply:
make sense.